src/App.vue

```diff
@@ -163,15 +164,48 @@ import MiniWidgetContainer from './components/MiniWidgetContainer.vue'
 import SlideToConfirm from './components/SlideToConfirm.vue'
 import Alerter from './components/widgets/Alerter.vue'
 import { datalogger } from './libs/sensors-logging'
+import { useMainVehicleStore } from './stores/mainVehicle'
 import { useWidgetManagerStore } from './stores/widgetManager'
 
 const widgetStore = useWidgetManagerStore()
+const vehicleStore = useMainVehicleStore()
 
 const showConfigurationMenu = ref(false)
 
 // Main menu
 const showMainMenu = ref(false)
 
+let requestDisarmConfirmationPopup = false
+const openMainMenu = (): void => {
+  if (vehicleStore.isArmed) {
+    requestDisarmConfirmationPopup = true
+    Swal.fire({
+      title: 'Be careful',
+      text: 'Vehicle is currently armed, its not recommended to open the main menu.',
+      icon: 'warning',
+      showCancelButton: true,
+      showConfirmButton: true,
+      cancelButtonText: 'Continue anyway',
+      confirmButtonText: 'Disarm vehicle',
+    }).then((result) => {
+      console.log('result', result)
+      if (result.isConfirmed && vehicleStore.isArmed) {
+        vehicleStore.disarm()
+      }
+      requestDisarmConfirmationPopup = false
+      showMainMenu.value = true
+    })
+  } else {
+    showMainMenu.value = true
+  }
+}
+const isVehicleArmed = computed(() => vehicleStore.isArmed)
+watch(isVehicleArmed, (isArmed) => {
+  if (requestDisarmConfirmationPopup && !isArmed) {
+    Swal.close()
+  }
+})
```

Comment:
I didn't get it on why to add this `requestDisarmConfirmationPopup` logic instead of always checking current state when the user tries to open the menu.

Reply:
About this I added because from what I know the Swal.close will close any pop up that is open on the moment, so I was thinking in a scenario where some other popup is open and the user interacts with the isArmed state it would probably close the popup that its not meant to be closed

Reply:
Got it!